### PR TITLE
CF-129: Need the ability to create one-time personalized discount codes

### PR DESCRIPTION
**Pull request by vlatal-gb**

### CODE REVIEW

File: server_extensions_api/src/main/java/com/generalbytes/batm/server/extensions/IExtensionContext.java [hunk 1]
```diff
@@ -493,4 +493,21 @@ IIdentity updateIdentity(String identityId, String externalId, int state, int ty
...
+     * @param validityFrom
+     * @param validityTill
+     * @param maximumUsages
+     * @param maximumUsagesPerIdentity
+     * @param minimumTransactionAmount
+     * @param totalLimit
+     * @param fiatCurrency in which currency are limit amounts
+     * @return Newly created Discount details.
+     */
+    IDiscount createIdentityLinkedDiscount(String identityPublicId, String code, BigDecimal buyDiscount, BigDecimal sellDiscount, Date validityFrom, Date validityTill, Long maximumUsages, Long maximumUsagesPerIdentity, BigDecimal minimumTransactionAmount, BigDecimal totalLimit, String fiatCurrency);
```
Comment (generalbytes) [inline server_extensions_api/src/main/java/com/generalbytes/batm/server/extensions/IExtensionContext.java:512]: 1. rename this method to createDiscount.
2. make String identityPublicId optional and move it as a last parameter.

File: gradle.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # buildscript - project id
 projectGroup=com.generalbytes.batm.public
-projectVersion=1.0.18-SNAPSHOT
+projectVersion=1.0.19
```
Comment (generalbytes) [inline gradle.properties:3]: Why did you delete -SNAPSHOT?

**Review by generalbytes — CHANGES_REQUESTED** [review]
Please implement required changes.

File: gradle.properties (hunk 2 above)
Comment (vlatal-gb) [reply gradle.properties:3]: We discussed with Filip, that the versions are not snapshots - those are releases, in fact.
